src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -6837,17 +6849,15 @@ final void TryProcessFeatureExtAck(boolean featureExtAckReceived) throws SQLServ
     }
 
     final void start() {
-        timerThread = new Thread(this);
-        timerThread.setDaemon(true);
-        timerThread.start();
+        task = executor.submit(this);
     }
 
     final void stop() {
+        task.cancel(false);
```

Comment:
This must be `cancel(true)` like in the original implementation (timerThread.interrupt was called). Else new threads will be started. E.g. if 10 queries are fired with a time-out within one second (which is the sleep-interval in the run-method), 10 threads will be created.